src/parser.js

```diff
@@ -1,35 +1,27 @@
-import q from 'q'
-import http from 'q-io/http'
+import Promise from 'bluebird'
+import requestAsync from 'request-promise'
 import validator from 'validator'
 import {default as podcastParser} from 'node-podcast-parser'
 
 class Parser {
 
-  static process(response) {
+  static process(xml) {
 
-    const deferred = q.defer()
-    podcastParser(response, (err, data) => {
-      if (err) {
-        deferred.reject(new Error(err))
-      } else
-        deferred.resolve(data)
+    return new Promise((res, rej) => {
+      podcastParser(xml, (err, data) => err ? rej(err) : res(data))
     })
-    return deferred.promise
 
   }
 
   static parse(url) {
 
-    if(!validator.isURL(url))
-      throw new Error('Invalid Podcast URL')
-
-    return http.read({
-      url: url,
-      method: 'GET'
-    }).then((response) => {
-      return response.toString()
-    }).then(Parser.process)
-
+    return new Promise((res, rej) => {
+      if (typeof url !== 'string' || !validator.isURL(url)) return rej(new Error('Invalid Podcast URL'))
```

Comment:
That's right, Bluebird works a little differently than I tought. In this case we just need to remove `typeof url !== 'string'` because validator already handle this.